app/controllers/ads_controller.rb

```diff
@@ -28,6 +28,30 @@ def create
     end
   end
 
+  def edit
+    @ad = Ad.active.find(params[:id])
+    redirect_to ad_path(@ad), alert: "Nedozvoljena akcija." if token_invalid?
+  end
+
+  def update
+    @ad = Ad.active.find(params[:id])
+
+    if token_valid? && @ad.update(ad_params)
+      redirect_to ad_path(@ad), notice: "Izmjene spremljene"
+    else
+      flash.now[:alert] = "Nedozvoljena akcija." if token_invalid?
```

Comment:
Validation errors will be displayed. This will take care of it https://github.com/vlado/earthquake-croatia/blob/42dcb0564dbd282244a31815c547644892b99cbe/app/views/ads/_form.html.erb#L5-L17